### PR TITLE
remove use of ldclient

### PR DESCRIPTION
**Pull request by jazanne**
We had partially switched to using ld client for requests, but it's easier for us to maintain one client, so I'm removing it and using the main http client for consistency.

Other
- add a helper for all api paths
- update to latest ldapi client `v15`, and latest api version `20220603`
- disable `nilnil` in linter

### CODE REVIEW

File: options/options.go [hunk 1]
```diff
@@ -260,7 +260,7 @@ func (o Options) Validate() error {
 	}
 
 	if o.Revision != "" && o.Branch == "" {
-		return fmt.Errorf(`"branch" option is required when "revision" option is set`)
+		return errors.New(`"branch" option is required when "revision" option is set`)
```
Comment (jazanne) [inline options/options.go:263]:
```suggestion
		return fmt.Errorf(`"branch" option is required when "revision" option is set`)
```

File: internal/ld/ld.go [hunk 2]
```diff
@@ -42,10 +40,10 @@ type ApiOptions struct {
 }
 
 const (
-	apiVersion       = "20210729"
+	apiVersion       = "20220603"
```
Comment (InTheCloudDan) [inline internal/ld/ld.go:43]: @jazanne this looks good to me, but I don't know the status of the API version now. Are we handling the pagination easily at this point?

**Review by InTheCloudDan — APPROVED** [review]

File: internal/ld/ld.go (hunk 2 above)
Comment (jazanne) [reply internal/ld/ld.go:43]: pagination still isn't required (forthcoming in "v4")

File: internal/ld/ld.go [hunk 3]
```diff
@@ -21,14 +20,13 @@ import (
 	h "github.com/hashicorp/go-retryablehttp"
 	"github.com/olekukonko/tablewriter"
 
-	ldapi "github.com/launchdarkly/api-client-go/v7"
+	ldapi "github.com/launchdarkly/api-client-go/v14"
```
Comment (mmrj) [inline internal/ld/ld.go:23]: do you want to switch to v15 while you're here?

**Review by mmrj — APPROVED** [review]
✅ from Docs to unblock you -- looks like we were tagged because there are `*.md` files in the `vendor` update, but I only looked at the non-`vendor` files. Let me know if there's anything you want specific Docs feedback on.

**Pull request merged**